app.js

```diff
@@ -201,7 +201,8 @@ function orchestrate(spec, isManifest) {
     spec.jobs['token-checker'].status = 'pending';
     var shortlink; //@todo retrieve shortlink
     return TokenChecker.check(shortlink, token).then(function(authReport) {
-      if(authReport.authorized) {
+      var matchSource = spec.url.substring(0, authReport.source.length) === spec.url;
```

Comment:
Shouldn't that check that `spec.url.substring(/* ... */) === authReport.source` instead? If not, I am not sure to understand what is being checked.